src/toffy/fov_watcher.py

```diff
@@ -479,6 +493,40 @@ def on_created(self, event: FileCreatedEvent, check_last_fov: bool = True):
             super().on_created(event)
             self._run_callbacks(event, check_last_fov)
 
+    def file_timer(self, fov_timeout, watcher_timeout):
+        """Checks time since last file was generated
+        Args:
+
+            fov_timeout (int):
+                how long to wait for fov data to be generated once file detected
+            watcher_timeout (int):
+                length to wait for new file generation before timing out
+        """
+        while True:
+            current_time = datetime.now()
+            time_elapsed = (current_time - self.last_event_time).total_seconds()
+
+            # 3 fov cycles and no new files --> timeout
+            if time_elapsed > watcher_timeout:
+                print("Timed out waiting for new file to be generated.")
```

Comment:
I'd add, specifically, which FOV it timed out waiting on files for (equivalent to `last_fov_num_updated + 1`).